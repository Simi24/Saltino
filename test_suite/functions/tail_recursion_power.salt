// Test di power/exponentiation con tail recursion
def main() {
    return power_tail(2, 10, 1)
}

def power_tail(base, exp, acc) {
    if (exp == 0) {
        return acc
    } else {
        return power_tail(base, exp - 1, acc * base)
    }
}
