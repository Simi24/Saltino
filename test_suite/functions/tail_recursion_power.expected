Program result: 1024
